Sources/TransifexNative/Locales.swift

```diff
@@ -9,30 +9,45 @@
 import Foundation
 
 /// Protocol that developers can use to create custom classes that return the current locale of the app.
-/// If no CurrentLocaleProvider is provided in LocaleState constructor, the UserDefaultsLocaleProvider
+/// If no CurrentLocaleProvider is provided in LocaleState constructor, the TXPreferredLocaleProvider
 /// is used.
 @objc
-public protocol CurrentLocaleProvider {
-
+public protocol TXCurrentLocaleProvider {
     func currentLocale() -> String
 }
 
-/// Class that returns the current locale found in the User Defaults dictionary of the app
-public final class UserDefaultsLocaleProvider : NSObject, CurrentLocaleProvider {
+/// Class that returns the language code of the current user's locale and falls back to "en" if the language
+/// code cannot be found.
+public final class TXPreferredLocaleProvider : NSObject {
+    private var _currentLocale : String
     
-    private static let APPLE_LANGUAGE_KEY = "AppleLanguages"
-    private static let FALLBACK_LANGUAGE = "en"
+    override init() {
+        // Fetch the current locale on initialization and return it when it's
+        // requested by the `currentLocale()` method.
+        _currentLocale = TXPreferredLocaleProvider.fetchCurrentLocale()
+
+        super.init()
+
+        // Detect whenever the current locale is changed and update that value.
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(currentLocaleDidChange),
+                                               name: NSLocale.currentLocaleDidChangeNotification,
+                                               object: nil)
+    }
+    
+    @objc
+    private func currentLocaleDidChange() {
+        _currentLocale = TXPreferredLocaleProvider.fetchCurrentLocale()
+    }
     
+    private static func fetchCurrentLocale() -> String {
```

Comment:
`fetch` sounds kind of "heavy". Is there a reason to avoid the simpler `get`?

Reply:
Renamed it to `getCurrentLocale()`